engine/src/mode_test.cpp

```diff
@@ -490,15 +494,13 @@ bool MCModeHandleMessageBoxChanged(MCExecContext& ctxt, MCStringRef)
 }
 
 // The standalone mode causes a relaunch message.
-bool MCModeHandleRelaunch(const char *& r_id)
+bool MCModeHandleRelaunch(MCStringRef& r_id)
 {
 #ifdef _WINDOWS
 	bool t_do_relaunch;
-	const char *t_id;
 
 	t_do_relaunch = MCdefaultstackptr -> hashandler(HT_MESSAGE, MCM_relaunch) == True;
-	t_id = MCdefaultstackptr -> getname_cstring();
-	r_id = t_id;
+	r_id = MCNameGetString(getname());
```

Comment:
Callers of `MCModeHandleRelaunch` appear to own the id string, so I think this needs a retain.